src/object/schedcontrol.c

```diff
@@ -102,7 +102,9 @@ static exception_t decodeSchedControl_Configure(word_t length, cap_t cap, extra_
     }
 
     time_t budget_us = mode_parseTimeArg(0, buffer);
+    ticks_t budget = usToTicks(budget_us);
     time_t period_us = mode_parseTimeArg(TIME_ARG_SIZE, buffer);
+    ticks_t period = usToTicks(period_us);
```

Comment:
If I remember correctly I deliberately did the comparisons on us to avoid overflow.

Reply:
I see. I think this change was made because it was difficult to determine the bound in terms of ticks on the verification side. I will discuss with verification, I could also do the check on both.

Reply:
The important thing to work out is what are the bounds you have now put on what these values can be. IIRC I chose these numbers to minimise restrictions on what the user could use and what we could check without overflowing. One way resulted in much more reduced values.

Reply:
It seems that would depend on the timer in use and whether it was more or less coarse than a microsecond. The minimum of both may end up being the effective upper bound anyway.

Reply:
I'm not sure this was practical for all of the systems we have seL4 ported to. Some of them are very high frequency.

Reply:
I think that because the ticks_t is 64bits wide, even with a 1terahertz clock rate `getMaxUsToTicks()` is over 500,000 years in duration. I can't think of a way that this loses any precision unless the clock speed is significantly less than 1 MHz which I don't think is too common.  I may be wrong though as this makes my head hurt

Reply:
I'm sure running sel4test on all platforms supported by MCS will reveal why this needed to be done.

Note: I realise that there was a deadline to try and get all of the MCS changes through this week, but this really shouldn't be done until *all* tests pass on all platforms. Is that the case?

Reply:
After going through all of the platforms we currently support, there are a couple of platforms with clock speeds around 1MHz order-of-magnitude with the smallest being 1.2MHz for the hikey and 1MHz for the hifive.  Changing the MIN_BUDGET comparison to use ticks instead of cycles should still work as there is always a distinct cycle value for every us value (Only while the timer frequency is > 1MHz).  So if we keep the getMaxUsToTicks() comparison in us, and never have platforms with a clock < 1MHz frequency then this change shouldn't change the observable behavior of this function.

Reply:
Although this doesn't seem to agree with this part of the commit message:
>  This is needed as the conversion is not guaranteed to preserve the properties of the checks due to possible truncation.

How does verification treat `usToTicks`?

Reply:
After more discussion internally, we believe that verifying the check in microseconds is unlikely to be an issue.  So this change isn't required for now.